unyt/unit_object.py

```diff
@@ -371,12 +371,26 @@ def __itruediv__(self, u):
         )
 
     def __rmul__(self, u):
+        if not getattr(u, "is_Unit", False):
+            data = np.array(u, subok=True)
+            if (
+                issubclass(data.__class__, _import_cache_singleton.ua)
+                and type(data) is not _import_cache_singleton.ua
+                and type(data) is not _import_cache_singleton.uq
+            ):
```

Comment:
Is there a reason to distinguish `data.__class__` and `type(data)` that I'm missing here ?
```suggestion
            data_type = type(data)
            if (
                issubclass(data_type, _import_cache_singleton.ua)
                and data_type is not _import_cache_singleton.ua
                and data_type is not _import_cache_singleton.uq
            ):
```